custom-checks/checkstyle/src/test/resources/checkstyle/thingHandlerSetLabelCheckTest/org.openhab.binding.extending.handler/ThingHandlerExtendingIndirectlyIncorrectBTH.java

```diff
@@ -0,0 +1,16 @@
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+
+public class ThingHandlerExtendingIndirectlyIncorrectBTH extends ThingHandlerWithSetLabel{
```

Comment:
Will this file get a warning, would it also get a warning if it extended `ThingHandlerWithThingFieldWithoutSetLabel`?

Reply:
No, it will not get a warning. For the first case the check will not even parse it because the ThingHandlerWithSetLabel class here extends incorrect BTH. And for the second case the check will create a new HashMap of files for the current directory: in this test case "org.openhab.binding.extending.handler" and ThingHandlerWithThingFieldWithoutSetLabel is from "org.openhab.binding.thing.handler"